### PR TITLE
Add MQ brokers resource

### PR DESCRIPTION
**Pull request by irmatov**
Please, take a look

### CODE REVIEW

**Comment by yevgenypats** [comment]
Closes #92

**Review by roneli — CHANGES_REQUESTED** [review]
Looks good, a few descriptions missing / broken and one missing column in users. I would also add TF integration tests to the MQ. @fdistorted can help you out if needed.

File: resources/mq_brokers.go [hunk 1]
```diff
@@ -0,0 +1,318 @@
...
+			{
+				Name:     "account_id",
+				Type:     schema.TypeString,
+				Resolver: client.ResolveAWSAccount,
+			},
+			{
+				Name:     "region",
+				Type:     schema.TypeString,
+				Resolver: client.ResolveAWSRegion,
+			},
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/mq_brokers.go]: Added

File: resources/mq_brokers.go [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+				Resolver: schema.PathResolver("EncryptionOptions.UseAwsOwnedKey"),
+			},
+			{
+				Name:     "encryption_options_kms_key_id",
+				Type:     schema.TypeString,
+				Resolver: schema.PathResolver("EncryptionOptions.KmsKeyId"),
+			},
+			{
+				Name:        "engine_type",
+				Description: "Required",
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/mq_brokers.go]: Fixed. Went over all of the descriptions and ensured all are present and correct.

File: resources/mq_brokers.go [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "host_instance_type",
+				Description: "The broker's instance type.",
+				Type:        schema.TypeString,
+			},
+			{
+				Name:        "publicly_accessible",
+				Description: "Required",
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/mq_brokers.go]: Fixed

File: resources/mq_brokers.go [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+			},
+			{
+				Name:        "tags",
+				Description: "The list of all tags associated with this broker.",
+				Type:        schema.TypeJSON,
+			},
+		},
+		Relations: []*schema.Table{
+			{
+				Name:         "aws_mq_broker_configurations",
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/mq_brokers.go:127]: Fixed

**Review by fdistorted — COMMENTED** [review]
Everything looks fine. Good job!
Take a look at Ron's suggestions.

File: resources/mq_brokers.go [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+				},
+			},
+			{
+				Name:         "aws_mq_broker_users",
+				Resolver:     fetchMqBrokerUsers,
+				Multiplex:    client.AccountRegionMultiplex,
+				IgnoreError:  client.IgnoreAccessDeniedServiceDisabled,
+				DeleteFilter: client.DeleteAccountRegionFilter,
+				Columns: []schema.Column{
+					{
```
[the remark this answers is not in the document]
Comment (irmatov) [reply resources/mq_brokers.go:276]: Added, along with more fields

File: docs/tables/cache_behaviour_lambda_function_associations.md [hunk 6]
```diff
@@ -0,0 +1,10 @@
+
```
Comment (roneli) [inline docs/tables/cache_behaviour_lambda_function_associations.md:1]: I am pretty sure this file shouldn't exist, if it does (find by table name) than that table should begin with aws_ prefix

**Review by roneli — APPROVED** [review]
Looks great. Just 1 minor comment to check if to delete that file or not and we can merge

**Review by roneli — APPROVED** [review]

**Pull request merged**